test/test_compile.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+import os.path
+import pytest
+
+from otk.command import compile
+
+
+@pytest.mark.parametrize(
+    "arguments, input_data, output_data, sys_exit_code, log_message",
+    [
+        # "GENERATE" is a placeholder to append "tmp_path" from pytest
+        (
+            argparse.Namespace(input="GENERATE", output="GENERATE", target=None),
+            """otk.version: 1
+otk.target.osbuild.qcow2: { test: 1 }
+""",
+            """{
+  "test": 1,
+  "version": "2",
+  "sources": {}
+}""",
+            0,
+            None,
+        ),
+    ],
+)
+def test_compile(
+    tmp_path,
+    caplog,
+    capsys,
+    monkeypatch,
+    arguments,
+    input_data,
+    output_data,
+    sys_exit_code,
+    log_message,
+):
+    if arguments.input == "GENERATE":
+        input_filename = "input.yaml"
+        arguments.input = os.path.join(tmp_path, input_filename)
```

Comment:
(super niptick) I like the way pathlib supports joins for paths ` arguments.input = tmp_path / input_filename` (but probably very personal and maybe does not work because input expects a string not a pathlib.Path?

Reply:
oh - yea - I'm just no used to this pathlib feature and still feels strange to override operators like this but I'll change